starport/pkg/clispinner/clispinner.go

```diff
@@ -23,31 +23,38 @@ func New() *Spinner {
 	s := &Spinner{
 		sp: sp,
 	}
-	s.SetText("Initializing...")
-	return s
+	return s.SetText("Initializing...")
 }
 
 // SetText sets the text for spinner.
 func (s *Spinner) SetText(text string) *Spinner {
+	s.sp.Lock()
 	s.sp.Suffix = " " + text
+	s.sp.Unlock()
 	return s
 }
 
 // SetPrefix sets the prefix for spinner.
 func (s *Spinner) SetPrefix(text string) *Spinner {
+	s.sp.Lock()
 	s.sp.Prefix = text + " "
+	s.sp.Unlock()
 	return s
 }
 
 // SetCharset sets the prefix for spinner.
 func (s *Spinner) SetCharset(charset []string) *Spinner {
+	s.sp.Lock()
```

Comment:
locking here probably will cause to a deadlock because UpdateCharSet also acquires a lock.

please see: https://github.com/briandowns/spinner/blob/v1.11.1/spinner.go#L386

Reply:
done